File: messenger/website/conversations.py
from flask import Blueprint, render_template, request, abort, jsonify, session, redirect, flash, g, url_for
from bcrypt import hashpw, checkpw, gensalt
from functools import wraps

from database import db, Conversation, User

conversations = Blueprint('conversations', __name__)

@conversations.route('', methods=['POST'])
def new_conversation():
    data = request.get_json()
    new_conversation = Conversation()
    fields = ['users']
    if not all(field in data for field in fields) or not all(data[field] for field in fields):
        return abort(401)
    for user_id in data['users']:
        user = User.query.get(user_id)
        if user:
            new_conversation.users.append(user)

    db.session.add(new_conversation)
    db.session.commit()

    return jsonify({
        'success': True,
        'conversation': new_conversation.lean(),
    })

@conversations.route('/<id>', methods=['GET', 'DELETE'])
def individual(id):
    conversation = Conversation.query.get_or_404(id)

    if request.method == 'GET':
        return conversation.serialize()

    if request.method == 'DELETE':
        conversation.delete()
        db.session.commit()

        return jsonify({
            'success': True,
        })

@conversations.route('/<id>/messages', methods=['GET'])
def messages(id):
    return jsonify([])
    conversation = Conversation.query.get_or_404(id)
    return jsonify([m.serialize() for m in conversation.messages()])

File: messenger/components/messages/actions/conversations.js
import axios from 'axios';

export const REQUEST_CONVERSATIONS = 'REQUEST-CONVERSATIONS';
export function requestConversations(){
  return {
    type: REQUEST_CONVERSATIONS,
  };
}

export const CONVERSATIONS_TO_STATE = 'CONVERSATIONS-TO-STATE';
export function conversationsToState(conversations) {
  return {
    type: CONVERSATIONS_TO_STATE,
    conversations,
  };
}

export function loadConversations() {
  return dispatch => {
    dispatch(requestConversations());
    axios.get('/conversations')
    .then(({data}) => {
      let conversations = {};
      data.map(conversation => {
        conversations[conversation.id] = conversation;
      });
      dispatch(conversationsToState(conversations));
    });
  };
}
